ravenframework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping.py

```diff
@@ -291,27 +298,33 @@ def FindRank(ref, x, xM):
 
       @ In, ref: array-like (1D), reference sample used for scaling data 'x'.
       @ In, x: array-like (2D), the matrix [samples x timesteps] for which the rank is to be determined.
-      @ In, xM: array-like (2D), the matrix [samples x timesteps] used to calculate measurement error.
+      @ In, recError, array-like (1D), reconstruction error to determine the rank of time series data.
       @ Out, xMAX[0]: rank, int, the rank of matrix 'x' that satisfies the condition on maximum residual.
       """
+
       x = (x.T/ref).T
-      error = (np.std(xM, axis=0)/ np.mean(xM, axis=0))[0] #msr error
+      # The precition of pcmTdep is sensitive to the rank here, which determines the dimension of
+      # U_exp, U_app subspaces for KNN regression. We will lose information under low rank, or
+      # overfit data with noise under high rank.
+      # This error can be made as user-defined.
+      if recError == None:
+        recError = 0.001 # 0.001 is decided as we want a rank corresponding to around 0.1% reconstruction error.
+      else:
+        recError = float(recError[0])
```

Comment:
Once you made the changes according to my previous comments, you do not need this lines, since recError becomes float instead of StringList.